### ISSUE TITLE
Why the software history was not kept?

### ISSUE THREAD

**Issue by gustavopinto** [opentoonz/opentoonz#640]
Hi there,

I'm a researcher studying software evolution. As part of my current research, I'm studying the implications of open-sourcing a proprietary software, for instance, if the project succeed in attracting newcomers. However, I observed that some projects, like _opentoonz_, deleted their software history.

https://github.com/opentoonz/opentoonz/commit/890dddabbd3346b06af2bcfa8d2402ead6bd00a9

Knowing that software history is indispensable for developers (e.g., developers need to refer to history several times a day), I would like to ask opentoonz developers the following four brief questions:

1. Why did you decide to not keep the software history?
2. Do the core developers faced any kind of problems, when trying to refer to the old history? If so, how did they solve these problems?
3. Do the newcomers faced any kind of problems, when trying to refer to the old history? If so, how did they solve these problems?
4. How does the lack of history impacted on software evolution? Does it placed any burden in understanding and evolving the software?

Thanks in advance for your collaboration,

Gustavo Pinto, PhD
http://www.gustavopinto.org

**Comment by gustavopinto** [comment]
Hi @ideasman42, do you have any comments?

**Comment by ideasman42** [comment]
@gustavopinto I wasn't involved with releasing the code,

However from similar situations I've seen - part of the process of opening the code may include removing code owned by 3rd parties (where whoever is open-sourcing the project doesn't own copyright on all of the code), in this case it can impractical to remove some libraries while keeping the history (though with the right tools it can be done).

**Comment by gustavopinto** [comment]
Hi @ideasman42, 

thanks once again for answering our research inquiries. We were able to collect 35 responses and we drafted a research paper with the results. The paper was submitted and accepted for the 14th International Conference on Open Source Systems (http://oss2018.org/). You can find the paper [here](http://gustavopinto.org/lost+found/oss2018.pdf). Hope you enjoy reading the paper!

Thanks again,

Gustavo

**Comment by RodneyBaker** [comment]
Very interesting read.

**Comment by ideasman42** [comment]
Interesting, though IMHO the conclusion misses the point that this depends a lot on the project, the legacy of some code bases may be extremely important to understanding existing code, other times less so _(though I'd argue it's almost always preferable to keep)_.

For Blender we did not have access to the original history (pre-2000's), and while I can only assume it would have been good to have, I don't think it hurt the project overall too badly.

When migrating from SVN to GIT we kept the history, and I can confirm the default tools here are difficult to use (default tools didn't finish after running for a week - on a fast computer). It took 2 weeks fulltime using reposurgeon (an alternative tool) to develop a set of rules to migrate - as well as support, fixes, features added by its key developers.
I can understand if a project might not have the resources to do this.
